lti_consumer/fields.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from opaque_keys.edx.keys import CourseKey
+
+log = logging.getLogger(__name__)
+
+
+def _strip_object(key):
+    """
+    Strips branch and version info if the given key supports those attributes.
+    """
+    if hasattr(key, 'version_agnostic') and hasattr(key, 'for_branch'):
+        return key.for_branch(None).version_agnostic()
+    else:
+        return key
+
+
+def _strip_value(value, lookup='exact'):
+    """
+    Helper function to remove the branch and version information from the given value,
+    which could be a single object or a list.
+    """
+    if lookup == 'in':
+        stripped_value = [_strip_object(el) for el in value]
+    else:
+        stripped_value = _strip_object(value)
+    return stripped_value
+
+
+# TODO: This code has been copies and pasted from openedx.core.djangoapps.xmodule_django
+# Need to figure out how can we access stuff from edx-platform.
```

Comment:
@nasthagiri @jmbowman , what is the preferred way if we need modules from edx-platform – like here, I needed `CourseKey` from openedx.